File: redsheet-portal/src/app/database-models/email-configuration.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class EmailConfiguration {
    constructor(emailConfiguration?: EmailConfiguration )  {
        let properties = [ "defaultCredentials", "deliveryMethod", "emailConfigurationId", "enableSSL", "host", "isActive", "password", "port", "userName",];
        for (let property of properties)
            if (emailConfiguration && emailConfiguration[property])
                this[property] = emailConfiguration[property];
    }
 
	defaultCredentials : boolean = false ;
 
    @maxLength(100)
	deliveryMethod : string =   undefined;
 
	emailConfigurationId : number =   0 ;
 
	enableSSL : boolean = false ;
 
    @required()
    @maxLength(100)
	host : string =   undefined;
 
	isActive : boolean = false ;
 
    @maxLength(100)
	password : string =   undefined;
 
    @range(1,2147483647)
	port : number =   undefined;
 
    @maxLength(200)
	userName : string =   undefined;


}

File: RedSheet.DbEntities/Models/EmailConfiguration.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("EmailConfiguration", Schema = "dbo")]
	[Serializable]
	public partial class EmailConfiguration  
	{

	public Nullable<bool> DefaultCredentials { get; set; } 
 
	[MaxLength(100)]
	public string DeliveryMethod { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int EmailConfigurationId { get; set; } 

	public Nullable<bool> EnableSSL { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string Host { get; set; } 

	public Nullable<bool> IsActive { get; set; } 
 
	[MaxLength(100)]
	public string Password { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int Port { get; set; } 
 
	[MaxLength(200)]
	public string UserName { get; set; } 

	public EmailConfiguration()
	{
    }

    }
}

